File: javascript/Mod4/4.03/test/DynamicLoadingTest.js
// filename: test/DynamicLoadingTest.js
//const path = require('path')
const assert = require('assert')
const DynamicLoadingPage = require('../pages/DynamicLoadingPage')
require('./spec_helper')

describe('Dynamic Loading', function() {
  this.timeout(30000)
  let dynamicLoading

  beforeEach(async function() {
      // path.delimiter + path.join(__dirname, '..', 'vendor')
      // process.env.PATH += vendorDirectory
      dynamicLoading = new DynamicLoadingPage(this.driver)
  })

  // afterEach(async function() {
  //   await driver.quit()
  // })

  it('hidden element', async function() {
    await dynamicLoading.loadExample('1')
    assert(
      await dynamicLoading.isFinishTextPresent(), true, 'Finish text not displayed'
    )

  })
  it('rendered element', async function() {
    await dynamicLoading.loadExample('2')
    assert(
      await dynamicLoading.isFinishTextPresent(), true, 'Finish text not displayed'
    )
  })

})

File: python/Mod3/3.07/tests/dynamic_loading_test.py
# filename: tests/dynamic_loading_test.py
import pytest
import os
from selenium import webdriver
from pages.dynamic_loading_page import DynamicLoadingPage


@pytest.fixture
def dynamic_loading(request):
    _chromedriver = os.path.join(os.getcwd(), 'vendor', 'chromedriver')
    if os.path.isfile(_chromedriver):
        driver_ = webdriver.Chrome(_chromedriver)

    else:
        driver_ = webdriver.Chrome()

    dynamic_loading = DynamicLoadingPage(driver_)
    def quit():
        driver_.quit()

    request.addfinalizer(quit)
    return dynamic_loading

def test_hidden_element(dynamic_loading):
    dynamic_loading.load_example("1")
    assert dynamic_loading.finish_text_present()

def test_rendered_element(dynamic_loading):
    dynamic_loading.load_example("2")
    assert dynamic_loading.finish_text_present()